lib/dealership.rb

```diff
@@ -0,0 +1,68 @@
+# require './lib/car'
+#
+#
+#
+class Dealership
+
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @total_val = 0
+    @inv_by_cost = []
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    @inventory.count > 1
+  end
+
+  def cars_by_make(make)
+    by_make = @inventory.select do |car|
+      car.make == make
+    end
+    return by_make
+  end
+
+  def total_value
+    # total = 0
+    @inventory.each do |car|
+      @total_val += car.total_cost
+    end
+    return @total_val
+  end
+
+  def details
```

Comment:
The problem with using @total_val here is that if you create a new Dealership and add cars but don't call the `total_value` method first @total_value won't be set and will be 0 when trying to access the details. For example:

@car_1 = Car.new("Ford Mustang", 1500, 36)
@car_2 = Car.new("Toyota Prius", 1000, 48)
@dealership = Dealership.new("Acme Auto", "123 Main Street")

@dealership.details == {"total_value" => 0, "address" => "123 Main Street"}

rather than 

@dealership.details == {"total_value" => 102000, "address" => "123 Main Street"}